### PR TITLE
Writing: change Infinite Scroll toggles to radio buttons

### PR DESCRIPTION
**Pull request by eliorivero**
radio buttons

### CODE REVIEW

File: client/my-sites/site-settings/theme-enhancements.jsx [hunk 1]
```diff
@@ -4,14 +4,19 @@
 import React, { Component, PropTypes } from 'react';
 import { localize } from 'i18n-calypso';
 import { connect } from 'react-redux';
+import get from 'lodash/get';
+import assign from 'lodash/assign';
```
Comment (oskosk) [inline client/my-sites/site-settings/theme-enhancements.jsx]: We are currently using the `import { assign } from 'lodash'` form for importing lodash methods. You can combine `{ get, assign }` in a single statement.

File: client/my-sites/site-settings/theme-enhancements.jsx [hunk 2]
```diff
@@ -30,9 +35,69 @@ class ThemeEnhancements extends Component {
 		handleAutosavingToggle: PropTypes.func.isRequired,
 		isSavingSettings: PropTypes.bool,
 		isRequestingSettings: PropTypes.bool,
-		fields: PropTypes.object,
+		fields: PropTypes.object
 	};
 
+	constructor( props ) {
+		super( props );
+		this.getInfiniteMode = this.getInfiniteMode.bind( this );
```
Comment (oskosk) [inline client/my-sites/site-settings/theme-enhancements.jsx]: These bindings in the constructor can be dropped by defining the methods that need to be bound to the component's scope with the following pattern :

```
updateInfiniteMode = ( radio ) => {
...
};

By using an arrow function you lock the scope of the function to the class that is being definied.

File: client/my-sites/site-settings/theme-enhancements.jsx [hunk 3]
```diff
@@ -30,9 +35,69 @@ class ThemeEnhancements extends Component {
...
+		}*/
+	}
+
+	/**
+	 * Translate Infinite Scroll module and option status into our three values for the options.
+	 *
+	 * @returns {string} Check the Infinite Scroll and its mode and translate into a string.
+	 */
+	getInfiniteMode() {
+		if ( ! get( this.props, [ 'infinite-scroll' ], '' ) ) {
```
Comment (oskosk) [inline client/my-sites/site-settings/theme-enhancements.jsx]: We can drop the `[ ]` arround `'infinite-scroll'` here

File: client/my-sites/site-settings/theme-enhancements.jsx [hunk 4]
```diff
@@ -30,9 +34,69 @@ class ThemeEnhancements extends Component {
...
+			return 'infinite_default';
+		}
+		if ( get( this.props, [ 'infinite_scroll' ], 'infinite-scroll' ) ) {
```
Comment (oskosk) [inline client/my-sites/site-settings/theme-enhancements.jsx]: We can drop the [ ] arround 'infinite-scroll' here too

File: client/my-sites/site-settings/theme-enhancements.jsx [hunk 5]
```diff
@@ -55,43 +119,55 @@ class ThemeEnhancements extends Component {
...
-						) )
-					}
-				</div>
+				<FormLabel>
+					<FormRadio
+						name="infinite_mode"
+						value="infinite_scroll"
+						checked={ 'infinite_scroll' === this.state.infinite_mode }
+						onChange={ this.handleInfiniteScroll }
+						disabled={ isRequestingSettings }
```
Comment (oskosk) [inline client/my-sites/site-settings/theme-enhancements.jsx]: This all the other radio inputs probably need to be disabled when saving settings too. 

`disabled={ isRequestingSettings || isSavingSettings }`

**Comment by tyxla** [comment]
I've addressed this one in #13341, which has just been merged. Sorry again for overlapping work on this one, @eliorivero

**Comment by eliorivero** [comment]
Good work @tyxla 👍

**Pull request closed without merging**